File: add/codesnippet/CPP/p-system.drawing.printin_18_1.cpp
public:
   void Printing()
   {
      try
      {
         streamToPrint = gcnew StreamReader( filePath );
         try
         {
            printFont = gcnew System::Drawing::Font( "Arial",10 );
            PrintDocument^ pd = gcnew PrintDocument;
            pd->PrintPage += gcnew PrintPageEventHandler(
               this, &Form1::pd_PrintPage );
            pd->PrinterSettings->PrinterName = printer;
            // Set the page orientation to landscape.
            pd->DefaultPageSettings->Landscape = true;
            pd->Print();
         }
         finally
         {
            streamToPrint->Close();
         }
      }
      catch ( Exception^ ex ) 
      {
         MessageBox::Show( ex->Message );
      }
   }

File: add/codesnippet/CSharp/p-system.drawing.printin_18_1.cs
 public void Printing()
 {
    try
    {
       streamToPrint = new StreamReader (filePath);
       try
       {
          printFont = new Font("Arial", 10);
          PrintDocument pd = new PrintDocument(); 
          pd.PrintPage += new PrintPageEventHandler(pd_PrintPage);
          pd.PrinterSettings.PrinterName = printer;
          // Set the page orientation to landscape.
          pd.DefaultPageSettings.Landscape = true;
          pd.Print();
       } 
       finally
       {
          streamToPrint.Close() ;
       }
    } 
    catch(Exception ex)
    { 
       MessageBox.Show(ex.Message);
    }
 }
 